app/recurring_job.go

```diff
@@ -409,6 +413,23 @@ func (job *Job) doSnapshot() (err error) {
 	return nil
 }
 
+func (job *Job) deleteSnapIfRetainNumExceeded(volume *longhornclient.Volume) error {
+	volumeAPI := job.api.Volume
+
+	collection, err := volumeAPI.ActionSnapshotList(volume)
+	if err != nil {
+		return err
+	}
+	jobSnapshots := filterSnapshotsWithLabel(collection.Data, types.RecurringJobLabel, job.labels[types.RecurringJobLabel])
+	if len(jobSnapshots) > job.retain {
```

Comment:
This seems to be included in `doSnapshotCleanup`. Can we directly and blindly call `job.doSnapshotCleanup(false)`?

Reply:
Yes, I think it could do that.
Removed the function `deleteSnapIfRetainNumExceeded` and used `job.doSnapshotCleanup(false)` instead.
Thanks.